vector/src/main/java/im/vector/app/features/settings/ignored/IgnoredUsersViewModel.kt

```diff
@@ -89,24 +90,14 @@ class IgnoredUsersViewModel @AssistedInject constructor(@Assisted initialState:
             )
         }
 
-        session.unIgnoreUserIds(listOf(action.userId), object : MatrixCallback<Unit> {
-            override fun onFailure(failure: Throwable) {
-                setState {
-                    copy(
-                            unIgnoreRequest = Fail(failure)
-                    )
-                }
-
-                _viewEvents.post(IgnoredUsersViewEvents.Failure(failure))
-            }
-
-            override fun onSuccess(data: Unit) {
-                setState {
-                    copy(
-                            unIgnoreRequest = Success(data)
-                    )
-                }
+        viewModelScope.launch {
+            val result = runCatching { session.unIgnoreUserIds(listOf(action.userId)) }
+            setState {
+                copy(
+                        unIgnoreRequest = result.fold(::Success, ::Fail)
+                )
             }
-        })
+            result.onFailure { _viewEvents.post(IgnoredUsersViewEvents.Failure(it)) }
```

Comment:
I think there is a mistake here, this will be called even in case of success, no?

Reply:
Ha, wasn't sure for second.
It only gets called if it's a failure.
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/on-failure.html

Reply:
Ok, my bad, I misread it. I thought it was a call to matrixCallback.onFailure(). You are right. I need a break :)